src/components/NewCommentForm.tsx

```diff
@@ -1,8 +1,85 @@
-import React from 'react';
+import React, { FormEvent, useState } from 'react';
+import cn from 'classnames';
+
+import { Comment } from '../types';
+
+type Props = {
+  selectedPostId: number;
+  onAddComment: (newComment: Omit<Comment, 'id'>) => Promise<void>;
+};
+
+export const NewCommentForm: React.FC<Props> = ({
+  selectedPostId,
+  onAddComment,
+}) => {
+  const [name, setName] = useState('');
+  const [hasName, setHasName] = useState(true);
+  const [email, setEmail] = useState('');
+  const [hasEmail, setHasEmail] = useState(true);
+  const [commentBody, setCommentBody] = useState('');
+  const [hasCommentBody, setHasCommentBody] = useState(true);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const handleSubmit = (event: FormEvent) => {
+    event.preventDefault();
+
+    if (!name) {
+      setHasName(false);
+    }
+
+    if (!email) {
+      setHasEmail(false);
+    }
+
+    if (!commentBody) {
+      setHasCommentBody(false);
+    }
+
+    if (!name || !email || !commentBody) {
+      return;
+    }
+
+    setIsLoading(true);
+
+    const newComment = {
+      postId: selectedPostId,
+      name: name.trim(),
+      email: email.trim(),
+      body: commentBody.trim(),
+    };
+
+    onAddComment(newComment)
+      .then(() => setCommentBody(''))
+      .finally(() => setIsLoading(false));
```

Comment:
Ensure that the `onAddComment` function handles errors appropriately. Currently, the `finally` block resets the loading state, but it might be beneficial to handle errors explicitly to inform the user if the comment submission fails.